cu29/src/config.rs

```diff
@@ -361,27 +372,35 @@ impl CuConfig {
             .collect()
     }
 
-    /// Convenience method to get all edges in the configuration graph.
-    #[allow(dead_code)]
-    pub fn get_all_edges(&self) -> Vec<Edge> {
-        self.graph
-            .edge_indices()
-            .map(|edge| {
-                let (source, target) = self.graph.edge_endpoints(edge).unwrap();
-                (
-                    source.index() as NodeId,
-                    target.index() as NodeId,
-                    self.graph[edge].clone(),
-                )
-            })
-            .collect()
-    }
-
     /// Adds an edge between two nodes/tasks in the configuration graph.
     /// msg_type is the type of message exchanged between the two nodes/tasks.
-    pub fn connect(&mut self, source: NodeId, target: NodeId, msg_type: &str) {
-        self.graph
-            .add_edge(source.into(), target.into(), msg_type.to_string());
+    pub fn connect(
```

Comment:
Considering users are rarely going to batch, maybe there should be a split between `connect` and `connect_batched`

There's also `store` but i dont think this is worth building a whole configuration type just for this unless you picture it being a heavy call-site for users with more configurations on the way. Sadly we can't pass in a `Cnx` struct since the nodes are indices here and `Cnx` has them as strings

Reply:
Actually, maybe `CnxConfig` would be fine. It'd be stored in a `Cnx` and then you can get some nice `Default()` semantics while making it

Reply:
This API was made because I wanted to use PetGraph but really did not want to expose it to the user as it is way more general than what copper needs (but its features for searching nodes etc are pretty good for the engine itself)

Let me split the 2 methods and rename that to sampler for now. Maybe we can think of a kind of builder pattern to simplify the API at some point, it is usually a good way to add "optionality" in Rust APIs it seems like